src/idlcxx/src/streamers.c

```diff
@@ -124,6 +124,8 @@ struct streams {
   idl_buffer_t key_move;
   idl_buffer_t key_max;
   size_t keys;
+  bool key_max_sz_unlimited;
+  bool max_sz_unlimited;
```

Comment:
Am I right that currently all types are printed to the buffer in one go (I believe that was one of your last PRs)? If so, I think this doesn't fit very well with that as whether a type is "unlimited" is per-type and so this introduces complexity.

Reply:
I think you reached the wrong conclusion here.
If the unlimited size flag is set, the contents for the (key_)max_size function are discarded and just replaced with setting the stream cursor to the "unlimited" position (SIZE_MAX), in stead of filling it with the usual operations relating to any members.
Upon completing a struct, union, or typedef, and completing the tree, all streaming function contents are flushed to the file and the unlimited size flags are reset to false. This way each constructed type's functions will not be influenced by the previous ones' max_size_unlimited property.